frontends/concrete-python/concrete/fhe/compilation/module_compiler.py

```diff
@@ -0,0 +1,568 @@
+            error = "Module can only be compiled with `composable` activated."
+            raise RuntimeError(error)
+
+        module_artifacts = (
+            module_artifacts
+            if module_artifacts is not None
+            else ModuleDebugArtifacts(list(self.functions.keys()))
+        )
+
+        dbg = DebugManager(configuration)
+
+        try:
+            # Trace and fuse the functions
+            for name, function in self.functions.items():
+                inputset = inputsets[name] if inputsets is not None else None
+                function_artifacts = module_artifacts.functions[name]
+                function.evaluate("Compiling", inputset, self.configuration, function_artifacts)
+                assert function.graph is not None
+                dbg.debug_computation_graph(name, function.graph)
+
+            # Convert the graphs to an mlir module
+            mlir_context = self.compilation_context.mlir_context()
+            graphs = {}
+            for name, function in self.functions.items():
+                if function.graph is None:
+                    error = "Expected graph to be set."
+                    raise RuntimeError(error)
+                graphs[name] = function.graph
+            mlir_module = GraphConverter(self.configuration).convert_many(graphs, mlir_context)
+            mlir_str = str(mlir_module).strip()
+            dbg.debug_mlir(mlir_str)
+            module_artifacts.add_mlir_to_compile(mlir_str)
+
+            # Debug some function informations
+            for name, function in self.functions.items():
+                dbg.debug_bit_width_constaints(name, function.graph)
+                dbg.debug_bit_width_assignments(name, function.graph)
+                dbg.debug_assigned_graph(name, function.graph)
+
+            # Compile to a module!
+            with dbg.debug_table("Optimizer", activate=dbg.show_optimizer()):
+                output = FheModule(graphs, mlir_module, self.compilation_context, configuration)
+                if isinstance(output.runtime, ExecutionRt):
+                    client_parameters = output.runtime.client.specs.client_parameters
+                    module_artifacts.add_client_parameters(client_parameters.serialize())
+
+            dbg.debug_statistics(output)
+
+        except Exception:  # pragma: no cover
+            # this branch is reserved for unexpected issues and hence it shouldn't be tested
+            # if it could be tested, we would have fixed the underlying issue
+
+            # if the user desires so,
+            # we need to export all the information we have about the compilation
+
+            if self.configuration.dump_artifacts_on_unexpected_failures:
+                module_artifacts.export()
+
+                # traceback_path = self.artifacts.output_directory.joinpath("traceback.txt")
+                # with open(traceback_path, "w", encoding="utf-8") as f:
+                #     f.write(traceback.format_exc())
```

Comment:
Why is this disabled?

Reply:
I thought it was like that `compiler.py`, but apparently, I did it myself . Will change !